### ISSUE TITLE
Persian Translation

### ISSUE THREAD

**Issue by kinghadi** [donnemartin/system-design-primer#110]
Hi :)
- Maintainer(s): @kinghadi
- Please check out the [Translations Contributing Guidelines.][1]
- Original translations thread: #28
- I'm working on https://github.com/kinghadi/system-design-primer

Interested in helping? Let us know!

[1]:https://github.com/donnemartin/system-design-primer/blob/master/CONTRIBUTING.md#translations

**Comment by donnemartin** [comment]
@kinghadi awesome thanks!

**Comment by SabinDiwan** [comment]
Good one. Upfront SAYS Maintainer(s): @hadisinaee
Code=decode.HAD-IS-IN-AEE.
Spelling check<=>>>... . ...<haddeesanaeeei
Thanks to the one whoz done ENOUGH ALREADY .=many thanks
Al QURAN:I AM WAT MY SLAVE THINKS I AM.

???RIGHT?

**Comment by SabinDiwan** [comment]
=ur limit is in ur eyes.U BE UR JUDGE.
Had e sanae .limitless blessing
HAD=u know
IS=
IN=inside
AEE=urdu>>>eyes

Result=lord WITHIN. Say thx often enough.(enough times) ====u wanna be forgiven.??

**Comment by reza-agahi** [comment]
Hello, I'm interested in helping.
What can I do?

**Comment by donnemartin** [comment]
@hadisinaee how can others help?

**Comment by hadisinaee** [comment]
@reza-agahi would you mind reviewing my translation?

**Comment by reza-agahi** [comment]
@hadisinaee Yes, of course. I will review it ASAP and share my comments with you :)

**Comment by reza-agahi** [comment]
@hadisinaee I just read the half of your translation and faced with these major issues:
- Perisan is a rtl language so it should be rtl!
- This is just a translation. You should not add/remove any sentences to it unless you can not convey the concept with literal translation.
- There are many mispelled words in the document.
- It is better not using conversational words and sentences like 'یاد بگیریم که چطوری ...'
- Don't use a mixture of english and perisan words like 'کامیونیتی' or 'اسکوپ' or 'کش'. If it has not a good translation in Persian, just include the English word.
- ِYou should not combine paragraphs in original text in one paragraph.
- Some important wrong translations: 
    - trade-off: it is 'مصالحه' and is not 'تعادل'.
    - consistency: it is better to use 'سازگاری' for consistency and not using 'یکپارچگی' that is reminiscent of integrity.
    - best effort: it is simply 'بهترین تلاش' and is not 'بهترین روش'.
    - mitigate: it is 'کاهش دادن' and is not 'بر طرف کردن'.

**Comment by hadisinaee** [comment]
@reza-agahi Thank you for your comments. Would you mind making a pull request on my branch? So we can discuss changes there.

Branch: https://github.com/hadisinaee/system-design-primer/tree/persian-translation

**Comment by shywn-mrk** [comment]
I would like to help, How can I start?

**Comment by hadisinaee** [comment]
Hi @shywn-mrk! Thanks for your interest! These are the ways I think you can contribute to this project (feel free to suggest yours):
- You can review my translation and improve the wording 
- I haven't updated the translation to the latest version, so it might be a good starting point to see if you can add the latest changes.

We can discuss the issues over the translation repo: https://github.com/hadisinaee/system-design-primer/tree/persian-translation

**Comment by shywn-mrk** [comment]
Hi @hadisinaee Thank you so much!
Sure thing, I can do both. I'll review the translations and let you know.

**Comment by esmaily** [comment]
Hi @hadisinaee thank your for translations . I interest help you for translations.
Your are wonderful man

**Comment by hadisinaee** [comment]
Hi @esmaily ! Thanks for your interest! These are the ways I think you can contribute to this project (feel free to suggest yours):

- You can review my translation and improve the wording
- I haven't updated the translation to the latest version, so it might be a good starting point to see if you can add the latest changes.

We can discuss the issues over the translation repo: https://github.com/hadisinaee/system-design-primer/tree/persian-translation

**Comment by mr-amirfazel** [comment]
hey there @hadisinaee . I'm interested in helping you at this job. if there's anything left to do, I'd be glad to be involved :) Let me know

**Comment by hadisinaee** [comment]
Hey @mr-amirfazel! Thank you! 

Please make a PR with your changes and we continue the discussion over there. 

https://github.com/hadisinaee/system-design-primer/tree/persian-translation

**Comment by mohammadi-com** [comment]
Salam @hadisinaee,

I hope you're doing well. Our offices were next together when you were in HodHod and I was in Torob, if you remember :)
I would like to contribute to this. could you please guide.

Thanks

**Comment by hadisinaee** [comment]
Salaam @mohammadi-com !

Aha! Yes, I remember you. Yes, such a small world :)

Thanks for your interest! I shared a link to a branch in a comment above—that’s my fork of the project. The process is pretty straightforward: fork the repo, make your changes, and then submit a PR.

Just a heads up, the repo hasn’t been updated in years, so there could be a lot that needs changing.

**Comment by Omid888** [comment]
A better translation for "trade-off" could be "بده‌بستان".
This service is great for finding persian synonyms: [vajehdan.com](url)

**Comment by hosni** [comment]
Hi @reza-agahi and @hadisinaee,
First, thank you for your work here.
I’ve read this issue and just wanted to share that I’ve heard the term "**سبک‌سنگین**" used for "**trade-off**," and I think it’s beautiful.
